[JHBuild] Add libjxl to moduleset

dpino opened the pull request:
#### e45a46fae1a59709b4d5dd5c043352129364d150
<pre>
[JHBuild] Add libjxl to moduleset
<a href="https://bugs.webkit.org/show_bug.cgi?id=258341">https://bugs.webkit.org/show_bug.cgi?id=258341</a>

Reviewed by Carlos Alberto Lopez Perez.

Changeset &apos;265024@main&apos; enabled JPEGXL by default in WebKitGTK and WPE.
This caused a build failure in systems such as Ubuntu 20.04, Ubuntu
22.04 and Debian 11, which do not provide &apos;libjxl&apos; as a system library.

To work around the issue, the failing build bots disabled USE_JPEGXL. By
adding &apos;libjxl&apos; to the JHBuild moduleset, USE_JPEGXL can be enabled
again.

* Tools/gtk/jhbuild.modules:
* Tools/gtk/patches/libjxl-add-cmake-flag-provision-dependencies.patch: Added.
* Tools/jhbuild/jhbuild-minimal.modules:
* Tools/jhbuild/patches/libjxl-add-cmake-flag-provision-dependencies.patch: Added.
* Tools/wpe/jhbuild.modules:
* Tools/wpe/patches/libjxl-add-cmake-flag-provision-dependencies.patch: Added.

Canonical link: <a href="https://commits.webkit.org/265394@main">https://commits.webkit.org/265394@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b0193cf95ba87014c8de0d99cd469505528564dd

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/10777 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/10997 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/11298 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/12426 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/10335 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/10792 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/13371 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/10974 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/13237 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/10937 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/11846 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/9064 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/12830 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/9140 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/9721 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/16978 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/10211 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/9873 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/13123 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/10348 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/8437 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/9506 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2577 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/13779 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/10209 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

clopez commented on Tools/jhbuild/jhbuild-minimal.modules:
```diff
@@ -224,4 +226,32 @@
+            tag="v0.6.1"
+            checkoutdir="libjxl-0.6.1"
+            repo="github.com">
+      <patch file="libjxl-add-cmake-flag-provision-dependencies.patch" strip="1"/>
```
In order to tell jhbuild to only build this dep if is not on the system we need to specify a version and a pkg-config filename. Otherwise jhbuild will build always the lib even when the system already has it.
The idea of the minimal jhbuild is to only build libraries is strictly needed.

Add this:
```
<cmake id="libjxl" ....
 <pkg-config>libjxl.pc</pkg-config>
 <branch module="libjxl/libjxl.git" version="0.6.1" module="....
```
The key parts is to specify a version on the branch entry and also add a pkg-config entry before it with the name of the .pc file

dpino committed:
[JHBuild] Add libjxl to moduleset
https://bugs.webkit.org/show_bug.cgi?id=258341

Reviewed by Carlos Alberto Lopez Perez.

Changeset '265024@main' enabled JPEGXL by default in WebKitGTK and WPE.
This caused a build failure in systems such as Ubuntu 20.04, Ubuntu
22.04 and Debian 11, which do not provide 'libjxl' as a system library.

To work around the issue, the failing build bots disabled USE_JPEGXL. By
adding 'libjxl' to the JHBuild moduleset, USE_JPEGXL can be enabled
again.

* Tools/gtk/jhbuild.modules:
* Tools/gtk/patches/libjxl-add-cmake-flag-provision-dependencies.patch: Added.
* Tools/jhbuild/jhbuild-minimal.modules:
* Tools/jhbuild/patches/libjxl-add-cmake-flag-provision-dependencies.patch: Added.
* Tools/wpe/jhbuild.modules:
* Tools/wpe/patches/libjxl-add-cmake-flag-provision-dependencies.patch: Added.

Canonical link: https://commits.webkit.org/265394@main
